{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a359fa5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "007692d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "GCF_910575425.1\n",
    "GCF_000374365.1\n",
    "GCF_000375405.1\n",
    "GCF_900155865.1\n",
    "GCF_000428125.1\n",
    "GCF_018390535.1\n",
    "GCF_018292205.1\n",
    "GCF_000186225.1\n",
    "GCF_008121405.1\n",
    "GCF_016901995.1\n",
    "GCF_002998535.1\n",
    "GCF_014196955.1\n",
    "GCF_001688725.2\n",
    "GCF_017309675.2\n",
    "GCF_000614125.1\n",
    "GCF_018292125.1\n",
    "GCF_900104585.1\n",
    "GCF_001405695.1\n",
    "GCF_015555825.1\n",
    "GCF_900128495.1\n",
    "GCF_900142015.1\n",
    "GCF_900130125.1\n",
    "GCF_020341675.1\n",
    "GCF_000154525.1\n",
    "GCF_000212915.1\n",
    "GCF_007341375.1\n",
    "GCF_014131755.1\n",
    "GCF_900241005.1\n",
    "GCF_009193295.2\n",
    "GCF_000513195.1\n",
    "GCF_016894205.1\n",
    "GCF_900128905.1\n",
    "GCF_900130135.1\n",
    "GCF_003865075.1\n",
    "GCF_900128455.1\n",
    "GCF_018292165.1\n",
    "GCF_900129065.1\n",
    "GCF_900108345.1\n",
    "GCF_018289135.1\n",
    "GCF_020091505.1\n",
    "GCF_902375225.1\n",
    "GCF_009193325.2\n",
    "GCF_008571395.1\n",
    "GCF_019930665.1\n",
    "GCF_002998435.1\n",
    "GCF_000195635.1\n",
    "GCF_902374815.1\n",
    "GCF_024793055.1\n",
    "GCF_016889925.1\n",
    "GCF_001314995.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "19f20ab4",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = '/home/nastya/Downloads/bacteroides_50_gff/ncbi-genomes-2023-02-14/'\n",
    "genes = ['dinB', 'recA', 'ruvA', 'polA', 'recN', 'nusA', 'uvrA', 'uvrB', 'dnaK']\n",
    "files = [path+i for i in os.listdir(path)]\n",
    "for gff in files:\n",
    "    with open(gff, \"r\") as file:\n",
    "        for lines in file:\n",
    "            line = file.readline()\n",
    "            for gene in genes:\n",
    "                if gene in line:\n",
    "                    with open('phyletic_pattern.txt', 'a') as out:\n",
    "                        a = gff.split('/')[6]\n",
    "                        name = a.split('_')[0]+'_'+a.split('_')[1]\n",
    "                        out.write(f\"{name}\\t{gene}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "010d1603",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>recN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>dnaK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>recA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>497 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                file  gene\n",
       "0    GCF_001314995.1  uvrA\n",
       "1    GCF_001314995.1  uvrB\n",
       "2    GCF_001314995.1  recN\n",
       "3    GCF_001314995.1  uvrA\n",
       "4    GCF_001314995.1  ruvA\n",
       "..               ...   ...\n",
       "492  GCF_000375405.1  dnaK\n",
       "493  GCF_000375405.1  recA\n",
       "494  GCF_000375405.1  uvrA\n",
       "495  GCF_000375405.1  uvrB\n",
       "496  GCF_000375405.1  ruvA\n",
       "\n",
       "[497 rows x 2 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv('phyletic_pattern.txt', sep='\\t')\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "34ba7bc8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>recN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>polA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>nusA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>dnaK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>recA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                file  gene\n",
       "0    GCF_001314995.1  uvrA\n",
       "1    GCF_001314995.1  uvrB\n",
       "2    GCF_001314995.1  recN\n",
       "4    GCF_001314995.1  ruvA\n",
       "5    GCF_001314995.1  polA\n",
       "..               ...   ...\n",
       "491  GCF_000375405.1  nusA\n",
       "492  GCF_000375405.1  dnaK\n",
       "493  GCF_000375405.1  recA\n",
       "495  GCF_000375405.1  uvrB\n",
       "496  GCF_000375405.1  ruvA\n",
       "\n",
       "[446 rows x 2 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data.drop_duplicates(subset=['file', 'gene'])\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0ea1a0ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>recN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_001314995.1</td>\n",
       "      <td>polA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>nusA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>dnaK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>recA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>GCF_000375405.1</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  id  gene\n",
       "0    GCF_001314995.1  uvrA\n",
       "1    GCF_001314995.1  uvrB\n",
       "2    GCF_001314995.1  recN\n",
       "4    GCF_001314995.1  ruvA\n",
       "5    GCF_001314995.1  polA\n",
       "..               ...   ...\n",
       "491  GCF_000375405.1  nusA\n",
       "492  GCF_000375405.1  dnaK\n",
       "493  GCF_000375405.1  recA\n",
       "495  GCF_000375405.1  uvrB\n",
       "496  GCF_000375405.1  ruvA\n",
       "\n",
       "[446 rows x 2 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data.rename(columns={'file':'id'})\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "7b64bda7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteroides_acidifaciens</td>\n",
       "      <td>GCF_910575425.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacteroides_gallinarum</td>\n",
       "      <td>GCF_000374365.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>GCF_000375405.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bacteroides_bouchesdurhonensis</td>\n",
       "      <td>GCF_900155865.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bacteroides_graminisolvens</td>\n",
       "      <td>GCF_000428125.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bacteroides_propionicigenes</td>\n",
       "      <td>GCF_018390535.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Bacteroides_caccae</td>\n",
       "      <td>GCF_018292205.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Bacteroides_helcogenes</td>\n",
       "      <td>GCF_000186225.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Bacteroides_pyogenes</td>\n",
       "      <td>GCF_008121405.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Bacteroides_caecigallinarum</td>\n",
       "      <td>GCF_016901995.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Bacteroides_heparinolyticus</td>\n",
       "      <td>GCF_002998535.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Bacteroides_reticulotermitis</td>\n",
       "      <td>GCF_014196955.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Bacteroides_caecimuris</td>\n",
       "      <td>GCF_001688725.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Bacteroides_humanifaecis</td>\n",
       "      <td>GCF_017309675.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Bacteroides_rodentium</td>\n",
       "      <td>GCF_000614125.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Bacteroides_cellulosilyticus</td>\n",
       "      <td>GCF_018292125.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Bacteroides_ihuae</td>\n",
       "      <td>GCF_900104585.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Bacteroides_salyersiae</td>\n",
       "      <td>GCF_001405695.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Bacteroides_clarus</td>\n",
       "      <td>GCF_015555825.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Bacteroides_ilei</td>\n",
       "      <td>GCF_900128495.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Bacteroides_stercorirosoris</td>\n",
       "      <td>GCF_900142015.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Bacteroides_congonensis</td>\n",
       "      <td>GCF_900130125.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Bacteroides_intestinalis</td>\n",
       "      <td>GCF_020341675.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Bacteroides_stercoris</td>\n",
       "      <td>GCF_000154525.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Bacteroides_coprosuis</td>\n",
       "      <td>GCF_000212915.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Bacteroides_koreensis</td>\n",
       "      <td>GCF_007341375.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Bacteroides_thetaiotaomicron</td>\n",
       "      <td>GCF_014131755.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Bacteroides_cutis</td>\n",
       "      <td>GCF_900241005.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Bacteroides_luhongzhouii</td>\n",
       "      <td>GCF_009193295.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Bacteroides_timonensis</td>\n",
       "      <td>GCF_000513195.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Bacteroides_eggerthii</td>\n",
       "      <td>GCF_016894205.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Bacteroides_luti</td>\n",
       "      <td>GCF_900128905.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Bacteroides_togonis</td>\n",
       "      <td>GCF_900130135.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Bacteroides_faecalis</td>\n",
       "      <td>GCF_003865075.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Bacteroides_mediterraneensis</td>\n",
       "      <td>GCF_900128455.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Bacteroides_uniformis</td>\n",
       "      <td>GCF_018292165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Bacteroides_faecichinchillae</td>\n",
       "      <td>GCF_900129065.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Bacteroides_ndongoniae</td>\n",
       "      <td>GCF_900108345.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Bacteroides_xylanisolvens</td>\n",
       "      <td>GCF_018289135.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Bacteroides_faecis</td>\n",
       "      <td>GCF_020091505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Bacteroides_neonati</td>\n",
       "      <td>GCF_902375225.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Bacteroides_zhangwenhongii</td>\n",
       "      <td>GCF_009193325.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Bacteroides_finegoldii</td>\n",
       "      <td>GCF_008571395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Bacteroides_nordii</td>\n",
       "      <td>GCF_019930665.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Bacteroides_zoogleoformans</td>\n",
       "      <td>GCF_002998435.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Bacteroides_fluxus</td>\n",
       "      <td>GCF_000195635.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Bacteroides_oleiciplenus</td>\n",
       "      <td>GCF_902374815.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Candidatus_Bacteroides_intestinigallinarum</td>\n",
       "      <td>GCF_024793055.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Bacteroides_fragilis</td>\n",
       "      <td>GCF_016889925.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>GCF_001314995.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       species               id\n",
       "0                     Bacteroides_acidifaciens  GCF_910575425.1\n",
       "1                       Bacteroides_gallinarum  GCF_000374365.1\n",
       "2                Bacteroides_propionicifaciens  GCF_000375405.1\n",
       "3               Bacteroides_bouchesdurhonensis  GCF_900155865.1\n",
       "4                   Bacteroides_graminisolvens  GCF_000428125.1\n",
       "5                  Bacteroides_propionicigenes  GCF_018390535.1\n",
       "6                           Bacteroides_caccae  GCF_018292205.1\n",
       "7                       Bacteroides_helcogenes  GCF_000186225.1\n",
       "8                         Bacteroides_pyogenes  GCF_008121405.1\n",
       "9                  Bacteroides_caecigallinarum  GCF_016901995.1\n",
       "10                 Bacteroides_heparinolyticus  GCF_002998535.1\n",
       "11                Bacteroides_reticulotermitis  GCF_014196955.1\n",
       "12                      Bacteroides_caecimuris  GCF_001688725.2\n",
       "13                    Bacteroides_humanifaecis  GCF_017309675.2\n",
       "14                       Bacteroides_rodentium  GCF_000614125.1\n",
       "15                Bacteroides_cellulosilyticus  GCF_018292125.1\n",
       "16                           Bacteroides_ihuae  GCF_900104585.1\n",
       "17                      Bacteroides_salyersiae  GCF_001405695.1\n",
       "18                          Bacteroides_clarus  GCF_015555825.1\n",
       "19                            Bacteroides_ilei  GCF_900128495.1\n",
       "20                 Bacteroides_stercorirosoris  GCF_900142015.1\n",
       "21                     Bacteroides_congonensis  GCF_900130125.1\n",
       "22                    Bacteroides_intestinalis  GCF_020341675.1\n",
       "23                       Bacteroides_stercoris  GCF_000154525.1\n",
       "24                       Bacteroides_coprosuis  GCF_000212915.1\n",
       "25                       Bacteroides_koreensis  GCF_007341375.1\n",
       "26                Bacteroides_thetaiotaomicron  GCF_014131755.1\n",
       "27                           Bacteroides_cutis  GCF_900241005.1\n",
       "28                    Bacteroides_luhongzhouii  GCF_009193295.2\n",
       "29                      Bacteroides_timonensis  GCF_000513195.1\n",
       "30                       Bacteroides_eggerthii  GCF_016894205.1\n",
       "31                            Bacteroides_luti  GCF_900128905.1\n",
       "32                         Bacteroides_togonis  GCF_900130135.1\n",
       "33                        Bacteroides_faecalis  GCF_003865075.1\n",
       "34                Bacteroides_mediterraneensis  GCF_900128455.1\n",
       "35                       Bacteroides_uniformis  GCF_018292165.1\n",
       "36                Bacteroides_faecichinchillae  GCF_900129065.1\n",
       "37                      Bacteroides_ndongoniae  GCF_900108345.1\n",
       "38                   Bacteroides_xylanisolvens  GCF_018289135.1\n",
       "39                          Bacteroides_faecis  GCF_020091505.1\n",
       "40                         Bacteroides_neonati  GCF_902375225.1\n",
       "41                  Bacteroides_zhangwenhongii  GCF_009193325.2\n",
       "42                      Bacteroides_finegoldii  GCF_008571395.1\n",
       "43                          Bacteroides_nordii  GCF_019930665.1\n",
       "44                  Bacteroides_zoogleoformans  GCF_002998435.1\n",
       "45                          Bacteroides_fluxus  GCF_000195635.1\n",
       "46                    Bacteroides_oleiciplenus  GCF_902374815.1\n",
       "47  Candidatus_Bacteroides_intestinigallinarum  GCF_024793055.1\n",
       "48                        Bacteroides_fragilis  GCF_016889925.1\n",
       "49                          Bacteroides_ovatus  GCF_001314995.1"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table = pd.read_csv('species_id.txt', sep='\\t')\n",
    "table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "22fdc7f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_910575425.1</th>\n",
       "      <td>Bacteroides_acidifaciens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000374365.1</th>\n",
       "      <td>Bacteroides_gallinarum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000375405.1</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900155865.1</th>\n",
       "      <td>Bacteroides_bouchesdurhonensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000428125.1</th>\n",
       "      <td>Bacteroides_graminisolvens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_018390535.1</th>\n",
       "      <td>Bacteroides_propionicigenes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_018292205.1</th>\n",
       "      <td>Bacteroides_caccae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000186225.1</th>\n",
       "      <td>Bacteroides_helcogenes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008121405.1</th>\n",
       "      <td>Bacteroides_pyogenes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016901995.1</th>\n",
       "      <td>Bacteroides_caecigallinarum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002998535.1</th>\n",
       "      <td>Bacteroides_heparinolyticus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014196955.1</th>\n",
       "      <td>Bacteroides_reticulotermitis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001688725.2</th>\n",
       "      <td>Bacteroides_caecimuris</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_017309675.2</th>\n",
       "      <td>Bacteroides_humanifaecis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000614125.1</th>\n",
       "      <td>Bacteroides_rodentium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_018292125.1</th>\n",
       "      <td>Bacteroides_cellulosilyticus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900104585.1</th>\n",
       "      <td>Bacteroides_ihuae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001405695.1</th>\n",
       "      <td>Bacteroides_salyersiae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_015555825.1</th>\n",
       "      <td>Bacteroides_clarus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900128495.1</th>\n",
       "      <td>Bacteroides_ilei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900142015.1</th>\n",
       "      <td>Bacteroides_stercorirosoris</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900130125.1</th>\n",
       "      <td>Bacteroides_congonensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_020341675.1</th>\n",
       "      <td>Bacteroides_intestinalis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000154525.1</th>\n",
       "      <td>Bacteroides_stercoris</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000212915.1</th>\n",
       "      <td>Bacteroides_coprosuis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_007341375.1</th>\n",
       "      <td>Bacteroides_koreensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014131755.1</th>\n",
       "      <td>Bacteroides_thetaiotaomicron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900241005.1</th>\n",
       "      <td>Bacteroides_cutis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_009193295.2</th>\n",
       "      <td>Bacteroides_luhongzhouii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000513195.1</th>\n",
       "      <td>Bacteroides_timonensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016894205.1</th>\n",
       "      <td>Bacteroides_eggerthii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900128905.1</th>\n",
       "      <td>Bacteroides_luti</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900130135.1</th>\n",
       "      <td>Bacteroides_togonis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003865075.1</th>\n",
       "      <td>Bacteroides_faecalis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900128455.1</th>\n",
       "      <td>Bacteroides_mediterraneensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_018292165.1</th>\n",
       "      <td>Bacteroides_uniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900129065.1</th>\n",
       "      <td>Bacteroides_faecichinchillae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900108345.1</th>\n",
       "      <td>Bacteroides_ndongoniae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_018289135.1</th>\n",
       "      <td>Bacteroides_xylanisolvens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_020091505.1</th>\n",
       "      <td>Bacteroides_faecis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_902375225.1</th>\n",
       "      <td>Bacteroides_neonati</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_009193325.2</th>\n",
       "      <td>Bacteroides_zhangwenhongii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008571395.1</th>\n",
       "      <td>Bacteroides_finegoldii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_019930665.1</th>\n",
       "      <td>Bacteroides_nordii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002998435.1</th>\n",
       "      <td>Bacteroides_zoogleoformans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000195635.1</th>\n",
       "      <td>Bacteroides_fluxus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_902374815.1</th>\n",
       "      <td>Bacteroides_oleiciplenus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024793055.1</th>\n",
       "      <td>Candidatus_Bacteroides_intestinigallinarum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016889925.1</th>\n",
       "      <td>Bacteroides_fragilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001314995.1</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    species\n",
       "id                                                         \n",
       "GCF_910575425.1                    Bacteroides_acidifaciens\n",
       "GCF_000374365.1                      Bacteroides_gallinarum\n",
       "GCF_000375405.1               Bacteroides_propionicifaciens\n",
       "GCF_900155865.1              Bacteroides_bouchesdurhonensis\n",
       "GCF_000428125.1                  Bacteroides_graminisolvens\n",
       "GCF_018390535.1                 Bacteroides_propionicigenes\n",
       "GCF_018292205.1                          Bacteroides_caccae\n",
       "GCF_000186225.1                      Bacteroides_helcogenes\n",
       "GCF_008121405.1                        Bacteroides_pyogenes\n",
       "GCF_016901995.1                 Bacteroides_caecigallinarum\n",
       "GCF_002998535.1                 Bacteroides_heparinolyticus\n",
       "GCF_014196955.1                Bacteroides_reticulotermitis\n",
       "GCF_001688725.2                      Bacteroides_caecimuris\n",
       "GCF_017309675.2                    Bacteroides_humanifaecis\n",
       "GCF_000614125.1                       Bacteroides_rodentium\n",
       "GCF_018292125.1                Bacteroides_cellulosilyticus\n",
       "GCF_900104585.1                           Bacteroides_ihuae\n",
       "GCF_001405695.1                      Bacteroides_salyersiae\n",
       "GCF_015555825.1                          Bacteroides_clarus\n",
       "GCF_900128495.1                            Bacteroides_ilei\n",
       "GCF_900142015.1                 Bacteroides_stercorirosoris\n",
       "GCF_900130125.1                     Bacteroides_congonensis\n",
       "GCF_020341675.1                    Bacteroides_intestinalis\n",
       "GCF_000154525.1                       Bacteroides_stercoris\n",
       "GCF_000212915.1                       Bacteroides_coprosuis\n",
       "GCF_007341375.1                       Bacteroides_koreensis\n",
       "GCF_014131755.1                Bacteroides_thetaiotaomicron\n",
       "GCF_900241005.1                           Bacteroides_cutis\n",
       "GCF_009193295.2                    Bacteroides_luhongzhouii\n",
       "GCF_000513195.1                      Bacteroides_timonensis\n",
       "GCF_016894205.1                       Bacteroides_eggerthii\n",
       "GCF_900128905.1                            Bacteroides_luti\n",
       "GCF_900130135.1                         Bacteroides_togonis\n",
       "GCF_003865075.1                        Bacteroides_faecalis\n",
       "GCF_900128455.1                Bacteroides_mediterraneensis\n",
       "GCF_018292165.1                       Bacteroides_uniformis\n",
       "GCF_900129065.1                Bacteroides_faecichinchillae\n",
       "GCF_900108345.1                      Bacteroides_ndongoniae\n",
       "GCF_018289135.1                   Bacteroides_xylanisolvens\n",
       "GCF_020091505.1                          Bacteroides_faecis\n",
       "GCF_902375225.1                         Bacteroides_neonati\n",
       "GCF_009193325.2                  Bacteroides_zhangwenhongii\n",
       "GCF_008571395.1                      Bacteroides_finegoldii\n",
       "GCF_019930665.1                          Bacteroides_nordii\n",
       "GCF_002998435.1                  Bacteroides_zoogleoformans\n",
       "GCF_000195635.1                          Bacteroides_fluxus\n",
       "GCF_902374815.1                    Bacteroides_oleiciplenus\n",
       "GCF_024793055.1  Candidatus_Bacteroides_intestinigallinarum\n",
       "GCF_016889925.1                        Bacteroides_fragilis\n",
       "GCF_001314995.1                          Bacteroides_ovatus"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table.set_index('id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3d45e9b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GCF_910575425.1': {'species': 'Bacteroides_acidifaciens'},\n",
       " 'GCF_000374365.1': {'species': 'Bacteroides_gallinarum'},\n",
       " 'GCF_000375405.1': {'species': 'Bacteroides_propionicifaciens'},\n",
       " 'GCF_900155865.1': {'species': 'Bacteroides_bouchesdurhonensis'},\n",
       " 'GCF_000428125.1': {'species': 'Bacteroides_graminisolvens'},\n",
       " 'GCF_018390535.1': {'species': 'Bacteroides_propionicigenes'},\n",
       " 'GCF_018292205.1': {'species': 'Bacteroides_caccae'},\n",
       " 'GCF_000186225.1': {'species': 'Bacteroides_helcogenes'},\n",
       " 'GCF_008121405.1': {'species': 'Bacteroides_pyogenes'},\n",
       " 'GCF_016901995.1': {'species': 'Bacteroides_caecigallinarum'},\n",
       " 'GCF_002998535.1': {'species': 'Bacteroides_heparinolyticus'},\n",
       " 'GCF_014196955.1': {'species': 'Bacteroides_reticulotermitis'},\n",
       " 'GCF_001688725.2': {'species': 'Bacteroides_caecimuris'},\n",
       " 'GCF_017309675.2': {'species': 'Bacteroides_humanifaecis'},\n",
       " 'GCF_000614125.1': {'species': 'Bacteroides_rodentium'},\n",
       " 'GCF_018292125.1': {'species': 'Bacteroides_cellulosilyticus'},\n",
       " 'GCF_900104585.1': {'species': 'Bacteroides_ihuae'},\n",
       " 'GCF_001405695.1': {'species': 'Bacteroides_salyersiae'},\n",
       " 'GCF_015555825.1': {'species': 'Bacteroides_clarus'},\n",
       " 'GCF_900128495.1': {'species': 'Bacteroides_ilei'},\n",
       " 'GCF_900142015.1': {'species': 'Bacteroides_stercorirosoris'},\n",
       " 'GCF_900130125.1': {'species': 'Bacteroides_congonensis'},\n",
       " 'GCF_020341675.1': {'species': 'Bacteroides_intestinalis'},\n",
       " 'GCF_000154525.1': {'species': 'Bacteroides_stercoris'},\n",
       " 'GCF_000212915.1': {'species': 'Bacteroides_coprosuis'},\n",
       " 'GCF_007341375.1': {'species': 'Bacteroides_koreensis'},\n",
       " 'GCF_014131755.1': {'species': 'Bacteroides_thetaiotaomicron'},\n",
       " 'GCF_900241005.1': {'species': 'Bacteroides_cutis'},\n",
       " 'GCF_009193295.2': {'species': 'Bacteroides_luhongzhouii'},\n",
       " 'GCF_000513195.1': {'species': 'Bacteroides_timonensis'},\n",
       " 'GCF_016894205.1': {'species': 'Bacteroides_eggerthii'},\n",
       " 'GCF_900128905.1': {'species': 'Bacteroides_luti'},\n",
       " 'GCF_900130135.1': {'species': 'Bacteroides_togonis'},\n",
       " 'GCF_003865075.1': {'species': 'Bacteroides_faecalis'},\n",
       " 'GCF_900128455.1': {'species': 'Bacteroides_mediterraneensis'},\n",
       " 'GCF_018292165.1': {'species': 'Bacteroides_uniformis'},\n",
       " 'GCF_900129065.1': {'species': 'Bacteroides_faecichinchillae'},\n",
       " 'GCF_900108345.1': {'species': 'Bacteroides_ndongoniae'},\n",
       " 'GCF_018289135.1': {'species': 'Bacteroides_xylanisolvens'},\n",
       " 'GCF_020091505.1': {'species': 'Bacteroides_faecis'},\n",
       " 'GCF_902375225.1': {'species': 'Bacteroides_neonati'},\n",
       " 'GCF_009193325.2': {'species': 'Bacteroides_zhangwenhongii'},\n",
       " 'GCF_008571395.1': {'species': 'Bacteroides_finegoldii'},\n",
       " 'GCF_019930665.1': {'species': 'Bacteroides_nordii'},\n",
       " 'GCF_002998435.1': {'species': 'Bacteroides_zoogleoformans'},\n",
       " 'GCF_000195635.1': {'species': 'Bacteroides_fluxus'},\n",
       " 'GCF_902374815.1': {'species': 'Bacteroides_oleiciplenus'},\n",
       " 'GCF_024793055.1': {'species': 'Candidatus_Bacteroides_intestinigallinarum'},\n",
       " 'GCF_016889925.1': {'species': 'Bacteroides_fragilis'},\n",
       " 'GCF_001314995.1': {'species': 'Bacteroides_ovatus'}}"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = table.set_index('id').to_dict(orient='index')\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9ec30322",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_dict={}\n",
    "for key, value in a.items():\n",
    "    new_dict.update({key:value['species']})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "2782dcf4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GCF_910575425.1': 'Bacteroides_acidifaciens',\n",
       " 'GCF_000374365.1': 'Bacteroides_gallinarum',\n",
       " 'GCF_000375405.1': 'Bacteroides_propionicifaciens',\n",
       " 'GCF_900155865.1': 'Bacteroides_bouchesdurhonensis',\n",
       " 'GCF_000428125.1': 'Bacteroides_graminisolvens',\n",
       " 'GCF_018390535.1': 'Bacteroides_propionicigenes',\n",
       " 'GCF_018292205.1': 'Bacteroides_caccae',\n",
       " 'GCF_000186225.1': 'Bacteroides_helcogenes',\n",
       " 'GCF_008121405.1': 'Bacteroides_pyogenes',\n",
       " 'GCF_016901995.1': 'Bacteroides_caecigallinarum',\n",
       " 'GCF_002998535.1': 'Bacteroides_heparinolyticus',\n",
       " 'GCF_014196955.1': 'Bacteroides_reticulotermitis',\n",
       " 'GCF_001688725.2': 'Bacteroides_caecimuris',\n",
       " 'GCF_017309675.2': 'Bacteroides_humanifaecis',\n",
       " 'GCF_000614125.1': 'Bacteroides_rodentium',\n",
       " 'GCF_018292125.1': 'Bacteroides_cellulosilyticus',\n",
       " 'GCF_900104585.1': 'Bacteroides_ihuae',\n",
       " 'GCF_001405695.1': 'Bacteroides_salyersiae',\n",
       " 'GCF_015555825.1': 'Bacteroides_clarus',\n",
       " 'GCF_900128495.1': 'Bacteroides_ilei',\n",
       " 'GCF_900142015.1': 'Bacteroides_stercorirosoris',\n",
       " 'GCF_900130125.1': 'Bacteroides_congonensis',\n",
       " 'GCF_020341675.1': 'Bacteroides_intestinalis',\n",
       " 'GCF_000154525.1': 'Bacteroides_stercoris',\n",
       " 'GCF_000212915.1': 'Bacteroides_coprosuis',\n",
       " 'GCF_007341375.1': 'Bacteroides_koreensis',\n",
       " 'GCF_014131755.1': 'Bacteroides_thetaiotaomicron',\n",
       " 'GCF_900241005.1': 'Bacteroides_cutis',\n",
       " 'GCF_009193295.2': 'Bacteroides_luhongzhouii',\n",
       " 'GCF_000513195.1': 'Bacteroides_timonensis',\n",
       " 'GCF_016894205.1': 'Bacteroides_eggerthii',\n",
       " 'GCF_900128905.1': 'Bacteroides_luti',\n",
       " 'GCF_900130135.1': 'Bacteroides_togonis',\n",
       " 'GCF_003865075.1': 'Bacteroides_faecalis',\n",
       " 'GCF_900128455.1': 'Bacteroides_mediterraneensis',\n",
       " 'GCF_018292165.1': 'Bacteroides_uniformis',\n",
       " 'GCF_900129065.1': 'Bacteroides_faecichinchillae',\n",
       " 'GCF_900108345.1': 'Bacteroides_ndongoniae',\n",
       " 'GCF_018289135.1': 'Bacteroides_xylanisolvens',\n",
       " 'GCF_020091505.1': 'Bacteroides_faecis',\n",
       " 'GCF_902375225.1': 'Bacteroides_neonati',\n",
       " 'GCF_009193325.2': 'Bacteroides_zhangwenhongii',\n",
       " 'GCF_008571395.1': 'Bacteroides_finegoldii',\n",
       " 'GCF_019930665.1': 'Bacteroides_nordii',\n",
       " 'GCF_002998435.1': 'Bacteroides_zoogleoformans',\n",
       " 'GCF_000195635.1': 'Bacteroides_fluxus',\n",
       " 'GCF_902374815.1': 'Bacteroides_oleiciplenus',\n",
       " 'GCF_024793055.1': 'Candidatus_Bacteroides_intestinigallinarum',\n",
       " 'GCF_016889925.1': 'Bacteroides_fragilis',\n",
       " 'GCF_001314995.1': 'Bacteroides_ovatus'}"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9c194b9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = data.replace({'id': new_dict})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "20ded50a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>uvrA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>recN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Bacteroides_ovatus</td>\n",
       "      <td>polA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>nusA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>dnaK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>recA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>uvrB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>Bacteroides_propionicifaciens</td>\n",
       "      <td>ruvA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                id  gene\n",
       "0               Bacteroides_ovatus  uvrA\n",
       "1               Bacteroides_ovatus  uvrB\n",
       "2               Bacteroides_ovatus  recN\n",
       "4               Bacteroides_ovatus  ruvA\n",
       "5               Bacteroides_ovatus  polA\n",
       "..                             ...   ...\n",
       "491  Bacteroides_propionicifaciens  nusA\n",
       "492  Bacteroides_propionicifaciens  dnaK\n",
       "493  Bacteroides_propionicifaciens  recA\n",
       "495  Bacteroides_propionicifaciens  uvrB\n",
       "496  Bacteroides_propionicifaciens  ruvA\n",
       "\n",
       "[446 rows x 2 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "b06a6dbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacteroides_acidifaciens</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_bouchesdurhonensis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_caccae</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_caecigallinarum</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_caecimuris</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_cellulosilyticus</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_clarus</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_congonensis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_coprosuis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_cutis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_eggerthii</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_faecalis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_faecichinchillae</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_faecis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_finegoldii</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_fluxus</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_fragilis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_gallinarum</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_graminisolvens</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_helcogenes</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_heparinolyticus</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_humanifaecis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_ihuae</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_ilei</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_intestinalis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_koreensis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_luhongzhouii</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_luti</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_mediterraneensis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_ndongoniae</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_neonati</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_nordii</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_oleiciplenus</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_ovatus</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_propionicifaciens</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_propionicigenes</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_pyogenes</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_reticulotermitis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_rodentium</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_salyersiae</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_stercorirosoris</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_stercoris</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_thetaiotaomicron</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_timonensis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_togonis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_uniformis</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_xylanisolvens</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_zhangwenhongii</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteroides_zoogleoformans</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Candidatus_Bacteroides_intestinigallinarum</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            gene\n",
       "id                                              \n",
       "Bacteroides_acidifaciens                       9\n",
       "Bacteroides_bouchesdurhonensis                 9\n",
       "Bacteroides_caccae                             9\n",
       "Bacteroides_caecigallinarum                    8\n",
       "Bacteroides_caecimuris                         9\n",
       "Bacteroides_cellulosilyticus                   9\n",
       "Bacteroides_clarus                             9\n",
       "Bacteroides_congonensis                        9\n",
       "Bacteroides_coprosuis                          9\n",
       "Bacteroides_cutis                              9\n",
       "Bacteroides_eggerthii                          9\n",
       "Bacteroides_faecalis                           9\n",
       "Bacteroides_faecichinchillae                   9\n",
       "Bacteroides_faecis                             9\n",
       "Bacteroides_finegoldii                         9\n",
       "Bacteroides_fluxus                             9\n",
       "Bacteroides_fragilis                           9\n",
       "Bacteroides_gallinarum                         9\n",
       "Bacteroides_graminisolvens                     8\n",
       "Bacteroides_helcogenes                         9\n",
       "Bacteroides_heparinolyticus                    8\n",
       "Bacteroides_humanifaecis                       9\n",
       "Bacteroides_ihuae                              9\n",
       "Bacteroides_ilei                               9\n",
       "Bacteroides_intestinalis                       9\n",
       "Bacteroides_koreensis                          9\n",
       "Bacteroides_luhongzhouii                       9\n",
       "Bacteroides_luti                               9\n",
       "Bacteroides_mediterraneensis                   9\n",
       "Bacteroides_ndongoniae                         9\n",
       "Bacteroides_neonati                            9\n",
       "Bacteroides_nordii                             9\n",
       "Bacteroides_oleiciplenus                       9\n",
       "Bacteroides_ovatus                             9\n",
       "Bacteroides_propionicifaciens                  9\n",
       "Bacteroides_propionicigenes                    9\n",
       "Bacteroides_pyogenes                           9\n",
       "Bacteroides_reticulotermitis                   9\n",
       "Bacteroides_rodentium                          8\n",
       "Bacteroides_salyersiae                         9\n",
       "Bacteroides_stercorirosoris                    9\n",
       "Bacteroides_stercoris                          9\n",
       "Bacteroides_thetaiotaomicron                   9\n",
       "Bacteroides_timonensis                         9\n",
       "Bacteroides_togonis                            9\n",
       "Bacteroides_uniformis                          9\n",
       "Bacteroides_xylanisolvens                      9\n",
       "Bacteroides_zhangwenhongii                     9\n",
       "Bacteroides_zoogleoformans                     9\n",
       "Candidatus_Bacteroides_intestinigallinarum     9"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.groupby('id').count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6917010b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
